config/schema.json

```diff
@@ -344,8 +344,89 @@
           }
         }
       }
-    }
+    },
+    "gateway": {
+      "description": "Configuration for the gateway proxy registration.",
+      "type": "object",
+      "properties": {
+        "kind": {
+          "description": "The gateway kind.",
+          "type": "string",
+          "enum": ["mesh-gateway", "terminating-gateway", "ingress-gateway"]
```

Comment:
Since we are working on mesh gateways first, what do you think about removing terminating and ingress gateway for now?